lib/oneview-sdk/resource/api300/c7000/enclosure_group.rb

```diff
@@ -26,11 +26,64 @@ def initialize(client, params = {}, api_ver = nil)
           # Default values:
           @data['type'] ||= 'EnclosureGroupV300'
           @data['stackingMode'] ||= 'Enclosure'
+          @data['enclosureCount'] ||= 1
           @data['interconnectBayMappingCount'] ||= 8
-          create_interconnect_bay_mapping unless @data['interconnectBayMappings']
           super
         end
 
+        # Adds the logical interconnect group
+        # @param [OneviewSDK::LogicalInterconnectGroup] lig Logical Interconnect Group
+        # @param [Integer] enclosureIndex Enclosure index of bay to add LIG to. If nil, interconnects will be added for all enclosures
+        # @raise [OneviewSDK::NotFound] if the LIG uri is not set and cannot be retrieved
+        # @return [OneviewSDK::API300::C7000::EnclosureGroup] self
+        def add_logical_interconnect_group(lig, enclosureIndex = nil)
+          lig.retrieve! unless lig['uri']
+          raise(NotFound, 'LIG not found!') unless lig['uri']
+          lig['interconnectMapTemplate']['interconnectMapEntryTemplates'].each do |entry|
+            entry['logicalLocation']['locationEntries'].each do |location|
+              next unless location['type'] == 'Bay' && entry['permittedInterconnectTypeUri']
+              add_lig_to_bay(location['relativeValue'], lig, enclosureIndex)
+            end
+          end
+          self
+        end
+
+        # Creates the interconnect bay mapping
+        # @return [OneviewSDK::API300::C7000::EnclosureGroup] self
+        def create_interconnect_bay_mapping
```

Comment:
See above. It still gets called (from inside the API200 initialize method), but this overrides the behavior of the method.